Godeps/_workspace/src/k8s.io/kubernetes/pkg/registry/generic/etcd/etcd.go

```diff
@@ -476,11 +476,9 @@ func (e *Etcd) WatchPredicate(ctx api.Context, m generic.Matcher, resourceVersio
 	}
 
 	if name, ok := m.MatchesSingle(); ok {
-		key, err := e.KeyFunc(ctx, name)
-		if err != nil {
-			return nil, err
+		if key, err := e.KeyFunc(ctx, name); err == nil {
+			return e.Storage.Watch(key, version, filterFunc)
```

Comment:
The scenario "do we match a single entry" is only valid when there is a valid key.  So if you get an error, it means you can't watch on the provided thing.